src/codegen.ts

```diff
@@ -273,24 +273,33 @@ function queryValues(
 ////////////////////////////////////////////////////////////////////////
 
 export type TsModule = {
-  sqlFileName: string
-  tsFileName: string
+  sqlFileName: string // full path
+  tsFileName: string // full path
   funcName: string
 }
 
 export type TsModuleDir = {
-  dirPath: string
+  dirPath: string // full path
+  nestedDirs: TsModuleDir[]
   modules: TsModule[]
   hasErrors: boolean
 }
 
-export function generateIndexModule(modules: TsModule[]): string {
-  return modules
+export function generateIndexModule(
+  dirPath: string,
+  nestedDirs: TsModuleDir[],
+  modules: TsModule[]
+): string {
+  const nestedDirsStr = nestedDirs
+    .map((dir) => `export * from '${path.relative(dirPath, dir.dirPath)}';`)
```

Comment:
I'm not sure it's the best way to "flatten" the directory structure by exporting like this. Maybe `export * as subdir from './subdir'` would be better? The you could `import * as sql from './sql'` and use `sql.subdir.func()` for using the queries in the subdirectory.

And BTW, even as is the code should be `export * from './${path...` because otherwise you're trying to import a package from `node_modules` :)